xarray/core/daskmanager.py

```diff
@@ -134,7 +134,7 @@ def map_blocks(
         *args: Any,
         dtype: np.typing.DTypeLike | None = None,
         chunks: tuple[int, ...] | None = None,
-        drop_axis: int | Sequence[int] | None = None,
+        drop_axis: int | Sequence[int] = [],
```

Comment:
Seems sketchy.

dask.array.map_blocks supports `drop_axis=None`: https://docs.dask.org/en/stable/_modules/dask/array/core.html#map_blocks

Reply:
Fair point. It seems the [actual default value in dask itself was changed](https://github.com/dask/dask/pull/9462) in dask version `2022.9.1`. It also appears older versions of dask don't support the new default value being passed. That's presumably why the [`min-all-deps` CI run failed here](https://github.com/pydata/xarray/actions/runs/4996474196/jobs/8949701978) - it used `dask-core=2022.1`. I'ved add a check for the dask version inside `map_blocks` and adjusted the passed parameter accordingly.